ReactAndroid/src/main/java/com/facebook/react/ReactFragment.java

```diff
@@ -0,0 +1,289 @@
+  }
+
+  /**
+   * @param componentName The name of the react native component
+   * @return A new instance of fragment ReactFragment.
+   */
+  private static ReactFragment newInstance(@NonNull String componentName, Bundle launchOptions) {
+    ReactFragment fragment = new ReactFragment();
+    Bundle args = new Bundle();
+    args.putString(ARG_COMPONENT_NAME, componentName);
+    args.putBundle(ARG_LAUNCH_OPTIONS, launchOptions);
+    fragment.setArguments(args);
+    return fragment;
+  }
+
+  // region Lifecycle
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    if (getArguments() != null) {
+      mComponentName = getArguments().getString(ARG_COMPONENT_NAME);
+      mLaunchOptions = getArguments().getBundle(ARG_LAUNCH_OPTIONS);
+    }
+    if (mComponentName == null) {
+      throw new IllegalStateException("Cannot loadApp if component name is null");
+    }
+    mDoubleTapReloadRecognizer = new DoubleTapReloadRecognizer();
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                           Bundle savedInstanceState) {
+    boolean needToEnableDevMenu = false;
+    if (getReactNativeHost().getUseDeveloperSupport() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      // Get permission to show redbox in dev builds.
+      if (!Settings.canDrawOverlays(getContext())) {
+        needToEnableDevMenu = true;
+        Intent serviceIntent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + getContext().getPackageName()));
+        FLog.w(ReactConstants.TAG, REDBOX_PERMISSION_MESSAGE);
+        Toast.makeText(getContext(), REDBOX_PERMISSION_MESSAGE, Toast.LENGTH_LONG).show();
+        startActivityForResult(serviceIntent, REQUEST_OVERLAY_CODE);
+      }
+    }
+    mReactRootView = createRootView();
+    if (!needToEnableDevMenu) {
+      mReactRootView.startReactApplication(
+        getReactNativeHost().getReactInstanceManager(),
+        mComponentName,
+        mLaunchOptions);
+    }
+    return mReactRootView;
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onHostResume(getActivity(), this);
+    }
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onHostPause(getActivity());
+    }
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    if (mReactRootView != null) {
+      mReactRootView.unmountReactApplication();
+      mReactRootView = null;
+    }
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onHostDestroy(getActivity());
+    }
+  }
+  // endregion
+
+  /**
+   * This currently only checks to see if we've enabled the permission to draw over other apps.
```

Comment:
But why is this method needed at all? Why receive a result from an activity? What activity? The parent?